apps/src/dance/ai/AiBlockPreview.tsx

```diff
@@ -0,0 +1,42 @@
+import {BlockSvg, Workspace, WorkspaceSvg} from 'blockly';
+import React, {useEffect, useRef} from 'react';
+import moduleStyles from './ai-block-preview.module.scss';
+
+interface AiPreviewProps {
+  generateBlocksFromResult: (workspace: Workspace) => [BlockSvg, BlockSvg];
+}
+
+/**
+ * Previews the blocks generated by the AI block in Dance Party.
+ */
+const AiBlockPreview: React.FunctionComponent<AiPreviewProps> = ({
+  generateBlocksFromResult,
+}) => {
+  const blockPreviewContainerRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (blockPreviewContainerRef.current) {
```

Comment:
do you mind explaining why we need this condition if it's assigned above?

Reply:
tbh I think I copied this pattern from the visualization preview -- I'm guessing it was there to protect against the ref being null on the initial render? But, since useEffect isn't called until after the first render (and, at that point, the ref points to a HTML element), I don't think it's necessary. Removed! (and tested locally)